lang/machine.md

```diff
@@ -41,27 +41,36 @@ struct StackFrame<M: Memory> {
     /// out-of-bounds index in the statement list), it refers to the terminator.
     next_stmt: Int,
 }
+
+struct CallerReturnInfo<M: Memory> {
+    /// The basic block to jump to when the callee returns.
+    /// If `None`, UB will be raised when the callee returns.
+    next_block: Option<BbName>,
+    /// The place where the caller wants to see the return value.
+    /// If `None`, the return value will be discarded.
+    place: Option<Place<M>>
```

Comment:
Codegen backends handle this using a dummy stack slot. Ignoring writes is a bit awkward as we allow for a return type `(u8, !)` to do `_0.1 = foo();`, which needs some room to store the return value and going the ignore route would require propagating the blackhole place to this call, which requires generating an extra copy of `foo` specifically for the case where the return value is ignored.

Reply:
@bjorn3 note that MiniRust differs from codegen/Miri in that on a call, we always allocate a return place for the callee. The caller-provided return place only becomes relevant when the function returns.

So using your terms, I think we always have such a "dummy stack slot". If the place here is `None`, that means the contents of that stack slot are discarded upon return; if it is `Some`, that means the contents are copied to the caller-provided place.